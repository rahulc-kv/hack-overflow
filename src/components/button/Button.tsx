import React, { FC, SVGProps } from 'react';

import useScreenDimensions from '@hooks/useScreenDimensions';

type ButtonProps = {
  className?: string;
  children?: React.ReactNode;
  onClick?: React.MouseEventHandler<HTMLButtonElement>;
  label?: string;
  labelClass?: string;
  iconClass?: string;
  Icon?: FC<SVGProps<SVGElement>>;
  disabled?: boolean;
  breakpoint?: number;
  type?: 'button' | 'submit' | 'reset';
  showMainViewButton?: boolean;
  iconStroke?: string;
  dataTestId?: string;
};

const Button: FC<ButtonProps> = ({
  className = '',
  onClick = () => null,
  label,
  labelClass,
  iconStroke = '',
  iconClass,
  Icon,
  disabled = false,
  breakpoint = 0,
  showMainViewButton = false,
  type = 'button',
  dataTestId = 'button'
}) => {
  const { isMobileView, screenWidth } = useScreenDimensions();

  return (
    <button
      type={type}
      className={`text-sm p-2 rounded-md hover:opacity-90 disabled:opacity-70 ${className}`}
      onClick={onClick}
      disabled={disabled}
      data-testid={dataTestId}>
      {(screenWidth < breakpoint || isMobileView) && !showMainViewButton ? (
        <div className="flex flex-row justify-center items-center">
          {Icon ? (
            <Icon stroke={iconStroke} />
          ) : (
            <p className={labelClass}>{label}</p>
          )}
        </div>
      ) : (
        <div className="flex flex-row justify-center items-center">
          {Icon && <Icon className={iconClass} stroke={iconStroke} />}
          {label && (
            <span className={`${Icon ? 'ml-2.5' : ''} ${labelClass}`}>
              {label}
            </span>
          )}
        </div>
      )}
    </button>
  );
};

export default Button;
