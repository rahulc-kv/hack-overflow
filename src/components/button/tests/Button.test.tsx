import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';

import Button from '../Button';

const handleClick = jest.fn();

test('Button Rendered', () => {
  const component = renderer.create(<Button onClick={handleClick} />).toJSON();
  expect(component).toMatchSnapshot();
});

test('Button Clicked', () => {
  render(<Button onClick={handleClick} />);
  fireEvent.click(screen.getByTestId('button'));
  expect(handleClick).toHaveBeenCalledTimes(1);
});
