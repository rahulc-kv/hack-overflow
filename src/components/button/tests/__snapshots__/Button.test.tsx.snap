// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Rendered 1`] = `
<button
  className="text-sm p-2 rounded-md hover:opacity-90 disabled:opacity-70 "
  data-testid="button"
  disabled={false}
  onClick={[MockFunction]}
  type="button"
>
  <div
    className="flex flex-row justify-center items-center"
  />
</button>
`;
